### PR TITLE
Capture and output subprocess stdout and stderr, and check subprocess execution return code

### PR DESCRIPTION
**Pull request by bnsblue**
# TL;DR
This PR aims to make flytekit_sagemaker_runner.py able to capture and out the subprocess's stdout and stderr. Also it captures the return code of the subprocess.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
This PR aims to make flytekit_sagemaker_runner.py able to capture and out the subprocess's stdout and stderr. Also it captures the return code of the subprocess.

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/__init__.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 import flytekit.plugins  # noqa: F401
 
-__version__ = "0.13.0b13"
+__version__ = "0.12.11"
```
Comment (wild-endeavor) [inline flytekit/__init__.py:5]: let's keep the minor and make a beta?  if you want to use 0.12.x can you make the same changes to the `release-0.12` branch?  Thanks.

**Review by wild-endeavor — APPROVED** [review]
